Src/WitsmlExplorer.Frontend/components/Modals/ServerModal.tsx

```diff
@@ -1,49 +1,111 @@
 import { DisplayModalAction, HideModalAction } from "../../contexts/operationStateReducer";
 import OperationType from "../../contexts/operationType";
 import { Server } from "../../models/server";
-import { BasicServerCredentials } from "../../services/credentialsService";
+import CredentialsService,{ BasicServerCredentials } from "../../services/credentialsService"
 import ServerService from "../../services/serverService";
 import { colors } from "../../styles/Colors";
-import ModalDialog from "./ModalDialog";
+import ModalDialog, { controlButtonPosition } from "./ModalDialog";
 import UserCredentialsModal, { CredentialsMode, UserCredentialsModalProps } from "./UserCredentialsModal";
+import { validText } from "./ModalParts";
+import WellService from "../../services/wellService";
+import NavigationContext from "../../contexts/navigationContext";
+import Icons from "../../styles/Icons";
+import NavigationType from "../../contexts/navigationType";
+import { useTheme } from "@material-ui/core/styles";
+import { CSSProperties } from "@material-ui/core/styles/withStyles";
 
 export interface ServerModalProps {
   server: Server;
-  dispatchNavigation: (action: AddServerAction | UpdateServerAction | RemoveWitsmlServerAction) => void;
+  dispatchNavigation: (action: AddServerAction | UpdateServerAction | RemoveWitsmlServerAction | UpdateWellsAction | SelectServerAction) => void;
   dispatchOperation: (action: HideModalAction | DisplayModalAction) => void;
   connectionVerified?: boolean;
+  serverCredentials?: BasicServerCredentials;
+  standalone?: boolean;
+  errorMessage?: string;
 }
 
 const ServerModal = (props: ServerModalProps): React.ReactElement => {
-  const { dispatchNavigation, dispatchOperation } = props;
+  const { dispatchOperation, serverCredentials } = props;
   const [server, setServer] = useState<Server>(props.server);
-  const [connectionVerified, setConnectionVerified] = useState<boolean>(props.connectionVerified ?? false);
+  const [connectionVerified, setConnectionVerified] = useState<boolean>(props.connectionVerified ?? true);
   const [displayUrlError, setDisplayUrlError] = useState<boolean>(false);
   const [displayNameError, setDisplayServerNameError] = useState<boolean>(false);
   const [isLoading, setIsLoading] = useState<boolean>(false);
-
   const isAddingNewServer = props.server.id === undefined;
   const schemeValues = ["Basic", "OAuth2"];
+  const [expanded, setExpanded] = useState<boolean>(false);
+  const [username, setUsername] = useState<string>();
+  const [password, setPassword] = useState<string>();
+  const [standAlone] = useState<boolean>(props.standalone ?? false);
+  const { dispatchNavigation } = props ?? useContext(NavigationContext);
+  const [errorMessage, setErrorMessage] = useState<string>("");
+
+  useEffect(() => {
+    if (serverCredentials) {
```

Comment:
We do not store username/password client side. Only when re-authenticating and caching credentials in backend. This probably will have to be discussed how to handle.